src/app/modules/shared-components/components/duration/duration.component.ts

```diff
@@ -1,41 +1,111 @@
-import { Component, EventEmitter, forwardRef, Input, Output } from '@angular/core';
-import { ControlValueAccessor, FormGroup, NG_VALUE_ACCESSOR } from '@angular/forms';
-import { Duration } from 'src/app/shared/helpers/duration';
-
-const MAX_HOURS_VALUE = 838;
+import { Component, EventEmitter, forwardRef, Injector, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  FormGroup,
+  NgControl,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+} from '@angular/forms';
+import { Duration, MAX_SECONDS_FORMAT_DURATION, MAX_TIME_FORMAT_DURATION } from 'src/app/shared/helpers/duration';
+
+const MAX_HOURS_VALUE = 23;
 const MAX_MINUTES_VALUE = 59;
 const MAX_SECONDS_VALUE = 59;
 
 @Component({
-  selector: 'alg-duration',
+  selector: 'alg-duration[ngModel], alg-duration[formControl], alg-duration[formControlName]',
   templateUrl: './duration.component.html',
   styleUrls: [ './duration.component.scss' ],
   providers: [
     {
       provide: NG_VALUE_ACCESSOR,
       useExisting: forwardRef(() => DurationComponent),
       multi: true,
-    }
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => DurationComponent),
+      multi: true,
+    },
   ]
 })
-export class DurationComponent implements ControlValueAccessor {
+export class DurationComponent implements OnInit, OnChanges, ControlValueAccessor, Validator {
   @Output() change = new EventEmitter<Duration | null>();
 
   @Input() name = '';
   @Input() parentForm?: FormGroup;
+  @Input() layout: 'DHM' | 'HMS' = 'HMS';
+  @Input() limitToTimeMax = false;
```

Comment:
My comment about that was

> Maybe just a limitToTimeMax = false would be more clear? (with a clear comment on what we consider "time" here.

So add "a clear comment on what we consider "time" here."